lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -171,25 +172,86 @@ module.exports = function(
   };
 
   /**
-   * @summary Get drive and image compatibility status object
+   * @summary Memoize ImmutableJS list reference
+   * @function
+   * @private
+   *
+   * @description
+   * This workaround is needed to avoid AngularJS from getting
+   * caught in an infinite digest loop when using `ngRepeat`
+   * over a function that returns a mutable version of an
+   * ImmutableJS object.
+   *
+   * The problem is that every time you call `myImmutableObject.toJS()`
+   * you will get a new object, whose reference is different from
+   * the one you previously got, even if the data is exactly the same.
+   *
+   * @param {Function} func - function that returns an ImmutableJS list
+   * @param {Function} compare - function used to compare the old and new lists
+   * @returns {Function} memoized function
+   *
+   * @example
+   * const getList = () => {
+   *   return Store.getState().toJS().myList;
+   * };
+   *
+   * const memoizedFunction = memoizeImmutableListReference(getList);
+   */
+  const memoizeImmutableListReference = (func, compare = _.isEqual) => {
+    let previous = [];
+
+    return (...restArgs) => {
+      const list = Reflect.apply(func, this, restArgs);
+
+      if (!_.isEqualWith(list, previous, compare)) {
+        previous = list;
+      }
+
+      return previous;
+    };
+  };
+
+  /**
+   * @summary Get the list of drives and their compatibility status objects
    * @function
    * @public
    *
    * @description
-   * Given an image and a drive, return their compatibility status object
-   * containing the status type (ERROR, WARNING, and OK), and accompanying
+   * For each drive, return its compatibility status with the seleted image,
+   * containing the status type (ERROR, WARNING), and accompanying
    * status message.
-   * Almost just a synonym of its namesake, the difference is that the image
-   * argument is implicit here for the sake of shortening.
    *
-   * @param {Object} drive - drive
-   * @returns {Object} compatibility status object
+   * @returns {Object[]} list of objects containing drive and statuses
    *
    * @example
-   * const { type, message } = DriveSelectorController.getDriveImageCompatibilityStatus(drive);
+   * const metadrives = DriveSelectorController.getDrivesAndStatuses();
+   *
+   * for ({ drive, statuses } of metadrives) {
+   *   for ({ type, message } of statuses) {
+   *     // do something
+   *   }
+   * }
    */
-  this.getDriveImageCompatibilityStatus = (drive) => {
-    return this.constraints.getDriveImageCompatibilityStatus(drive, this.state.getImage());
-  };
+  this.getDrivesAndStatuses = memoizeImmutableListReference(() => {
+    const statusList = [];
+
+    for (const drive of this.drives.getDrives()) {
+      statusList.push({
+        drive,
+        statuses: this.constraints.getDriveImageCompatibilityStatuses(drive, this.state.getImage())
+      });
+    }
+
+    return statusList;
+  }, (newList, oldList) => {
+    if (newList.length !== oldList.length) {
```

Comment:
We can remove it now, it's a relic of the block since I previously used `zipWith` which doesn't truncate in Lodash, however it is slightly more efficient when kept because it only checks the length at `O(1)` versus the structure at `O(n)`, but `n` is usually lower than the tens here so it might not be much of a problem...